crates/rslint_errors/src/emit.rs

```diff
@@ -120,21 +120,23 @@ impl<'files> Emitter<'files> {
 impl Emitter<'_> {
     /// Render and emit the diagnostic to stderr
     pub fn emit_stderr(&mut self, d: &Diagnostic, color: bool) -> Result<(), Error> {
-        let mut out = StandardStream::stderr(if color {
+        let out = StandardStream::stderr(if color {
             ColorChoice::Always
         } else {
             ColorChoice::Never
         });
+        let mut out = out.lock();
```

Comment:
Maybe we should document this implicit locking behavior?

Reply:
Yea that would be helpful.